From fc0718e3c0908d06aaa3ca0211841168399a797d Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Tue, 23 Nov 2010 18:10:41 -0500
Subject: [PATCH 010/625] update ltt-events: call ltt_session_destroy when
 ltt_events_exit

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Signed-off-by: Jianchuan Wang <jianchuan.wang@windriver.com>
[Rename patch subject to avoid name conflict]
Signed-off-by: Yang Shi <yang.shi@windriver.com>
---
 drivers/staging/lttng2/ltt-events.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/staging/lttng2/ltt-events.c b/drivers/staging/lttng2/ltt-events.c
index 688d7f3..96c90c4 100644
--- a/drivers/staging/lttng2/ltt-events.c
+++ b/drivers/staging/lttng2/ltt-events.c
@@ -150,10 +150,23 @@ static int __init ltt_events_init(void)
 	events_cache = KMEM_CACHE(ltt_event, 0);
 	if (!events_cache)
 		return -ENOMEM;
+
+	/* TODO: show ABI to userspace */
+
 	return 0;
 }
 
 static void __exit ltt_events_exit(void)
 {
+	struct ltt_session *session, *tmpsession;
+
+	/* TODO: hide ABI from userspace, wait for callers to release refs. */
+
+	list_for_each_entry_safe(session, tmpsession, &sessions, list)
+		ltt_session_destroy(session);
 	kmem_cache_destroy(events_cache);
 }
+
+MODULE_LICENSE("GPL and additional rights");
+MODULE_AUTHOR("Mathieu Desnoyers <mathieu.desnoyers@efficios.com>");
+MODULE_DESCRIPTION("LTTng Events");
-- 
2.0.2

